cmd/jujud/agent/machine.go

```diff
@@ -1198,8 +1203,12 @@ func (a *MachineAgent) apiserverWorkerStarter(
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		w, err := a.newAPIserverWorker(st, certChanged, dependencyReporter)
+		statePool := state.NewStatePool(st)
+		w, err := a.newAPIserverWorker(
+			st, statePool, certChanged, dependencyReporter,
+		)
 		if err != nil {
+			statePool.Close()
```

Comment:
should state pool close also close any states in the pool, save having to close any managed states explicitly?